src/boot/efi/boot.c

```diff
@@ -182,6 +182,9 @@ static BOOLEAN line_edit(
                         cursor_color = TEXT_ATTR_SWAP(cursor_color);
 
                         err = console_key_read(&key, 750 * 1000);
+                        if (!IN_SET(err, EFI_SUCCESS, EFI_TIMEOUT, EFI_NOT_READY))
+                                return FALSE;
```

Comment:
hmm, can you elaborate on the commit msg? how could we possibly end up with an uninitialized `key`? I mean, we'd leave this loop only on EFI_SUCCESS, but in that case the `key` should definitely be initialized... what am i missing?

Reply:
It's mostly theoretical and it doesn't really happen in this invocation. But the one in menu_run and print_status could. I'm just trying to be thorough on all of them.

Reply:
(Also, this check here is to prevent infinite looping for unexpected/fatal input errors)